Display category and date of the pages in list and content fragments

mpourismaiel opened the pull request:
**What this PR does / why we need it**:
Display date and categories in both content and list fragment

**Which issue this PR fixes**:
fixes #323 
fixes #332

**Release note**:
```release-note
- list: Display date and category of pages
- content: Display last modification date
```

mpourismaiel commented on layouts/partials/helpers/publish-date.html:
```diff
@@ -0,0 +1,8 @@
+<span class="text-muted mr-2">
+  {{- if ne $.root.Params.Lastmod $.root.Params.Date -}}
```
I seriously have no idea. I copied over the code and was working on it for 15 minutes to get it to work :D I'll see if it has any effect.

stp-ip commented on layouts/partials/helpers/publish-date.html:
```diff
@@ -0,0 +1,8 @@
+    {{- i18n "last_updated" | default "Last updated:" -}}
+  {{- end -}}
+  <time datetime="{{ $.root.Date }}" itemprop="datePublished dateModified">
+    {{- $.root.Lastmod.Format ($.site.Params.date_format | default "2006 January 26") -}}
```
That might actually be a weird Go Template issue so. But yeah weird.

mpourismaiel commented on layouts/partials/helpers/publish-date.html:
```diff
@@ -0,0 +1,8 @@
+<span class="text-muted mr-2">
+  {{- if ne $.root.Params.Lastmod $.root.Params.Date -}}
+    {{- i18n "last_updated" | default "Last updated:" -}}
```
I removed this section entirely. With the new code, `.Lastmod` is used by default and if there is `date` available in frontmatter, that is used instead.

mpourismaiel commented:
Extracted date to it's own partial, refactored it and changed date format to ISO. Also changed the styling to badge and disabled displaying date by default.
